### PR TITLE
fix(BaseStreamingImageVolume) Issue with typescript type

### LINKED ISSUE #807: [Bug] Cannot import cornerstoneStreamingImageVolumeLoader
### Describe the Bug

I'm trying to import **cornerstoneStreamingImageVolumeLoader** in my project and I get this error:
`
Error: node_modules/@cornerstonejs/streaming-image-volume-loader/dist/esm/StreamingImageVolume.d.ts:24:29 - error TS2307: Cannot find module 'packages/core/dist/esm/enums/RequestType' or its corresponding type declarations.

24         requestType: import("packages/core/dist/esm/enums/RequestType").default;
`

Versions:
    "@cornerstonejs/core": "^1.16.6",
    "@cornerstonejs/dicom-image-loader": "^1.16.6",
    "@cornerstonejs/streaming-image-volume-loader": "^1.16.6",
    "@cornerstonejs/tools": "^1.16.6",

### Steps to Reproduce

Import cornerstoneStreamingImageVolumeLoader with:
`import { cornerstoneStreamingImageVolumeLoader } from '@cornerstonejs/streaming-image-volume-loader';`

### The current behavior

Get an error
Seems related to the way it's declared (streaming-image-volume-loader.api.md):
`
import { default as default_2 } from 'packages/core/dist/esm/enums/RequestType';
`

### The expected behavior

Should work without error

### OS

Windows 11

### Node version

18.16.0

### Browser

NA

### PR DESCRIPTION
**Pull request by frolic06**
### Context

Fixes #807 : not able to import from streaming-image-volume-loader

### Changes & Results

Change the type of requestType from enum to string

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

- [x] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [x] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Review by sedghi — COMMENTED** [review]
I read about the bug your reported. and I'm surprised on the solution. Do you know what is happening? what are you using? vite? webpack?

**Comment by frolic06** [comment]
I'm using angular, so webpack. It seems to me that the enum value 'Enums.RequestType.Prefetch' is not see as the enum type Enums.RequestType, although the enum is correctly exported. That's why I change the type to string. It's probably not the cleaner solution, nor really elegant, but it seems to work.
If you know a better way to fix this, I will change my PR.

**Comment by frolic06** [comment]
requestType is already defined as a string here for example:
`interface ImageLoaderOptions {
    additionalDetails?: Record<string, unknown>;
    priority: number;
    requestType: string;
}`

**Comment by frolic06** [comment]
@sedghi I can also change with, if you prefer:
`const requestType: any = Enums.RequestType.Prefetch;`

There must be another solution, but I haven't found it. This solution has the advantage of being simple and removing unnecessary enum exports.

**Comment by sedghi** [comment]
what is the type declaration for other Enums that we have, are they strings?

**Comment by frolic06** [comment]
I searched the code history and found that the bug was introduced by PR #438 . The difference is that we have a class that inherits from another class. This seems to be the source of the problem, because the signature of  one of the method is created from an inherited private method.

**Comment by frolic06** [comment]
My solution is therefore to explicitly add an interface for the return of one of the methods

**Comment by frolic06** [comment]
@sedghi is this ok for you ?

File: packages/streaming-image-volume-loader/src/ImageLoadRequests.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Types, Enums } from '@cornerstonejs/core';
+
+export default interface ImageLoadRequests {
+  callLoadImage: (
+    imageId: any,
```
Comment (sedghi) [inline packages/streaming-image-volume-loader/src/ImageLoadRequests.ts:5]: string?

File: packages/streaming-image-volume-loader/src/ImageLoadRequests.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    imageIdIndex: any,
```
Comment (sedghi) [inline packages/streaming-image-volume-loader/src/ImageLoadRequests.ts:6]: number?

File: packages/streaming-image-volume-loader/src/ImageLoadRequests.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    options: any
+  ) => Promise<void>;
+  imageId: string;
+  imageIdIndex: number;
+  options: {
+    targetBuffer: {
+      arrayBuffer: SharedArrayBuffer;
+      offset: number;
+      length: number;
+      type: any;
```
Comment (sedghi) [inline packages/streaming-image-volume-loader/src/ImageLoadRequests.ts:16]: string?

File: packages/streaming-image-volume-loader/src/ImageLoadRequests.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  callLoadImage: (
+    imageId: any,
+    imageIdIndex: any,
+    options: any
+  ) => Promise<void>;
+  imageId: string;
+  imageIdIndex: number;
+  options: {
+    targetBuffer: {
+      arrayBuffer: SharedArrayBuffer;
```
Comment (sedghi) [inline packages/streaming-image-volume-loader/src/ImageLoadRequests.ts:13]: I think if we don't use the sharedArrayBuffer this is undefine so it is SharedArrayBuffer | undefined

**Comment by frolic06** [comment]
@sedghi fixed, anything else ?
Thanks for the review

**Review by sedghi — APPROVED** [review]

**Pull request merged**